File: src/main/java/org/mifos/fineract/services/LikelihoodApi.java
package org.mifos.fineract.services;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.PUT;
import rx.Observable;

public interface LikelihoodApi {
    /**
     * @param likelihoodId (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("likelihood/{ppiName}/{likelihoodId}")
    Observable<String> retrieve(
            @retrofit2.http.Path("likelihoodId") Long likelihoodId
    );

    /**
     * @param ppiName (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("likelihood/{ppiName}")
    Observable<String> retrieveAll(
            @retrofit2.http.Path("ppiName") String ppiName
    );

    /**
     * @param likelihoodId (required)
     * @param body         (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @PUT("likelihood/{ppiName}/{likelihoodId}")
    Observable<String> update(
            @retrofit2.http.Path("likelihoodId") Long likelihoodId, @retrofit2.http.Body String body
    );

}

File: src/main/kotlin/org/openapitools/client/apis/LikelihoodApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PUT
import retrofit2.http.Path


interface LikelihoodApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param likelihoodId
     * @param ppiName
     * @return [kotlin.String]
     */
    @GET("v1/likelihood/{ppiName}/{likelihoodId}")
    suspend fun retrieve(
        @Path("likelihoodId") likelihoodId: Long,
        @Path("ppiName") ppiName: String
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param ppiName
     * @return [kotlin.String]
     */
    @GET("v1/likelihood/{ppiName}")
    suspend fun retrieveAll11(@Path("ppiName") ppiName: String): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param likelihoodId
     * @param ppiName
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/likelihood/{ppiName}/{likelihoodId}")
    suspend fun update4(
        @Path("likelihoodId") likelihoodId: Long,
        @Path("ppiName") ppiName: String,
        @Body body: String? = null
    ): String

}
